### DISCUSSION TITLE
Workflow Feedback

### DISCUSSION THREAD

**Discussion by Extarys** [photoprism/photoprism#1327] [Ideas]
I'd like a way to add geolocations to my pictures from inside photoprism.
**This issue kind of gets off topic, maybe we need to change the title. Maybe "Workflow experience and confusion"?**
Also, I have trouble organizing my thoughts when I type, I apologize for the mess, I tried :stuck_out_tongue_closed_eyes: 

**Initial issue:**
What I started with:
I created an album called "Cellphone pictures" and uploaded a bunch of pictures there. Then assign some of them to other albums and tried to manually add geolocation data and other tags for more precise search.

The first idea that comes to mind for geotagging:
A way to **select a bunch of pictures and enter lat/lon once**, **select a location from a map**, or do it by **zipcode** or something for that. I believe it would be faster that the current workflow - see below.

I currently have a text file with coordinates and location name and I go through the pictures and add them manually, hence the second issue below.

I do know batch editing is on it's way (hopefully??) it's just quite painful to manually tag each individual picture with keywords, locations, etc. Not sure how other people are managing this.

**EDIT Second issue/Idea/...**
I was thinking about having a side bar with a map like shown on https://github.com/photoprism/photoprism/issues/271#issuecomment-613582727 to select location when clicking an Edit button of some sort. A sidebar would also be more intuitive to modify photo tags. Right now there is a lot of steps that need to be taken to modify a tag on 5 different pictures - not necessary batch edit, maybe just adding different keywords to different pictures (or different locations), I find the experience cumbersome. And I really don't mean this as a way to bash on the contributors and developers of photoprism.

How I currently do things - as a new user. My photo archive is a 10 year mess of folders inside folders
I created a couple of basic albums: Cellphone photos, apartments work, My cats, My aquariums

**Current workflow for tagging:**

1. Select a picture
2. Click Edit
3. Click Labels tab
4. Enter a label
5. Click +
6. Close or click next

Clicking next, there is no way to know on which picture you are on, except if you click the Details tab again before going back to Labels tab)

**Current workflow for geolocation:**

1. Select picture
2. Click Edit
3. Copy latitude from textfile
4. Click Latitude field and paste
5. Copy longitude from textfile
6. Click longitude field and paste
7. Click Apply
8. Click the Arrow until you find the other picture you want or, close the edit modal and select another picture

**Current workflow for sorting already uploaded pictures in different albums:**

1. Enter an album (in my case my Cellphone photos
2. Select some pictures of cats
3. Move them to the My cats album
4. Reselect all previously selected pictures from cellphone photos album
5. Delete them from current album

I believe having the option to display a sidebar with the current photo details would improve that workflow.

Now a bunch of newbie first-world problems I found when I started importing my files with the web upload tool.

**Albums/Calendar**
As a newbie, there is also the confusion of **Albums** and **Calendar** in the sidebar. Albums > Folders is the same as Calendar.
Reading the manual, it says I can put folders directly in my **originals** folder and it will show up as a Folder in the Albums > Folders screen. In future versions, eventually, I would suggest that first time users see some tips to explain how they can use photoprism to organize their photos. I think I'll use that for my huge memes collection.

**Albums/Tags confusion or limitation**
I'm overwhelmed by all the possibilities PhotoPrism offer. The lack of mass editing or the length I have to go through to edit tags makes me feel incompetent. I don't know if I should create an album for each event (christmas 2009), or tag "Christmas" on all the pictures of that event. Do I put the same picture in 4 different albums (Christmas, Family pictures, Parties, Jannet's House), add a bunch of tags and people's names (while face tagging is being worked on).. I think once geotagging and batch edit is implemented I won't feel as limited. I like sorting by date/event, location and subject. My feeling my disappear once I feel like I can do what I want.

**Small UX issue**
Another point, when you select an Album or a Folder and click **Edit** on the floating menu, there is the **Category** field, but I didn't find the purpose of that category field. The modal that appears when clicking on the Edit button of a Folder have the title: **Edit Album**, same as when you edit an album. This fuels the confusion voice in my head. :joy: 

The Category field also confuses me with the Labels - which one should I use when? Since all software are different, and I also never used one before, this is confusing - lots of that in this post, sorry again.

**Troubling Moments**
Importing a bunch of pictures also led to the creation of a **Moment** named Cats, which contain about 20 pictures (out of 1000 cat pictures). It contains some 2018 pictures and some 2019 pictures. It also appears that I cannot add pictures to that moment. From my understanding, Moments are auto-generated albums that are static? (cannot be modified?) And so I'm "stuck" with an incomplete moment?

I will also start scanning boxes of old photos soon once I get my Epson V600 and I wanted to get the hang of this before messing with some precious photos that will need thorough categorizing and sorting, hence this post.

I know this project is still a little baby and more work is needed and **I'm not bashing against the devs and contributors work** :heart: Keep in mind I'm not a photographer either and not a pro at organizing my files, I just enjoy my cats too much and the delete button is my enemy, therefore... lots of photos :D

**Comment by lastzero** [comment]
Yes, this should certainly be be part of batch edit. Until it's done, you may use other apps to set coordinates and re-index. Users owning external GPS trackers reported that this works well for them.

**Comment by Extarys** [comment]
@lastzero Okidoo thanks. I was editing the issue to cover more the workflow of things instead of just geolocation though. Can I go ahead with my edit? I wanted to explain how, as a new user sorting 10 years of picture mess, I was doing things in photoprism. I understand some users are professionals, but I wanted to use this as an all-in-one solution for a family guy who wants to sort and share his library.

**Comment by lastzero** [comment]
Sure, we love learning about use cases & pain points 🤗

**Comment by Extarys** [comment]
@lastzero I edited the post - If you want to rename this after, no problemo. I went over all the little and big things that came to mind after using PhotoPrism for 2-3 hours. I also read the documentation but I may have missed some stuff in there or misunderstood some ways of doing things. The batch edit is also in there and all but I know it's coming. I didn't mean it in a bad evil way.
